File: EPRO.Library.v3.5/EPRO.Library.v3.5/Modules/basExtensions.cs
using ELibrary.Standard.Objects;
using Microsoft.VisualBasic;

namespace ELibrary.Standard.Modules
{
    public static class basExtensions
    {

        #region MathsFunctions

        /// <summary>
        /// Toggles Integer Values e.g betw
        /// </summary>
        /// <param name="current"></param>
        /// <param name="min"></param>
        /// <param name="max"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static int ToggleValue(this int current, int min, int max)
        {
            if (min > max || current != max & current != min)
                return min;
            return min + (max - current);
        }


        #endregion


        #region Object Conversions
        // REM Dont put object as extension

        public static int toInt32(this object val)
        {
            return EInt.valueOf(val);
        }

        public static long toLong(this object val)
        {
            return ELong.valueOf(val);
        }

        public static double toDouble(this object val)
        {
            return EDouble.valueOf(val);
        }

        public static decimal toDecimal(this object val)
        {
            return EDecimal.valueOf(val);
        }

        /// <summary>
        /// Converts an Any Object to Short Type
        /// </summary>
        /// <param name="val"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static short toShort(this object val)
        {
            return EShort.valueOf(obj: val);
        }

        /// <summary>
        /// Converts an Any Object to Boolean
        /// </summary>
        /// <param name="val"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static bool toBoolean(this object val)
        {
            return EBoolean.valueOf(obj: val);
        }

        #endregion


        #region String Functions

        /// <summary>
        /// Checks if String1 equals String2 Ignoring the Case Sensitivity
        /// </summary>
        /// <param name="str1"></param>
        /// <param name="str2"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static bool equalsIgnoreCase(this string str1, string str2)
        {
            return EStrings.equalsIgnoreCase(str1, str2);
        }


        /// <summary>
        /// Converts to Proper Case
        /// </summary>
        /// <param name="pStr"></param>
        /// <returns></returns>
        /// <remarks></remarks>
        public static string toProperCase(this string pStr)
        {
            return Strings.StrConv(pStr, VbStrConv.ProperCase);
        }


        #endregion

    }
}

File: EPRO.Library.v3.5/EPRO.Library.v3.5/Modules/basExtensions.vb
Option Explicit On
Option Strict On

Imports EPRO.Library.v3._5.Objects

Namespace Modules



    Public Module basExtensions

#Region "MathsFunctions"

        ''' <summary>
        ''' Toggles Integer Values e.g betw
        ''' </summary>
        ''' <param name="current"></param>
        ''' <param name="min"></param>
        ''' <param name="max"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        <System.Runtime.CompilerServices.Extension()> _
        Public Function ToggleValue(ByVal current As Integer, ByVal min As Integer, ByVal max As Integer) As Integer
            If min > max OrElse (current <> max And current <> min) Then Return min

            Return min + (max - current)

        End Function


#End Region


#Region "Object Conversions"
        REM Dont put object as extension

        <System.Runtime.CompilerServices.Extension()> _
        Public Function toInt32(ByVal val As Object) As Int32
            Return EInt.valueOf(val)
        End Function


        <System.Runtime.CompilerServices.Extension()> _
        Public Function toLong(ByVal val As Object) As Long
            Return ELong.valueOf(val)
        End Function


        <System.Runtime.CompilerServices.Extension()> _
        Public Function toDouble(ByVal val As Object) As Double
            Return EDouble.valueOf(val)
        End Function


        <System.Runtime.CompilerServices.Extension()> _
        Public Function toDecimal(ByVal val As Object) As Decimal
            Return EDecimal.valueOf(val)
        End Function

        ''' <summary>
        ''' Converts an Any Object to Short Type
        ''' </summary>
        ''' <param name="val"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        <System.Runtime.CompilerServices.Extension()> _
        Public Function toShort(ByVal val As Object) As Short
            Return EShort.valueOf(obj:=val)
        End Function

        ''' <summary>
        ''' Converts an Any Object to Boolean
        ''' </summary>
        ''' <param name="val"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        <System.Runtime.CompilerServices.Extension()> _
        Public Function toBoolean(ByVal val As Object) As Boolean
            Return EBoolean.valueOf(obj:=val)
        End Function

#End Region


#Region "String Functions"

        ''' <summary>
        ''' Checks if String1 equals String2 Ignoring the Case Sensitivity
        ''' </summary>
        ''' <param name="str1"></param>
        ''' <param name="str2"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        <System.Runtime.CompilerServices.Extension()> _
        Public Function equalsIgnoreCase(ByVal str1 As String, ByVal str2 As String) As Boolean
            Return EStrings.equalsIgnoreCase(str1, str2)
        End Function


        ''' <summary>
        ''' Converts to Proper Case
        ''' </summary>
        ''' <param name="pStr"></param>
        ''' <returns></returns>
        ''' <remarks></remarks>
        <System.Runtime.CompilerServices.Extension()> _
        Public Function toProperCase(ByVal pStr As String) As String
            Return Strings.StrConv(pStr, VbStrConv.ProperCase)
        End Function


#End Region

    End Module

End Namespace